modules/puppetca/puppetca_puppet_cert.rb

```diff
@@ -39,6 +47,44 @@ def list
       hash.merge(ca_inventory) {|key, h1, h2| h1[:state] == "pending" ?  h1 : h1.merge(h2)}
     end
 
+    def list_puppet_6
+      find_puppetca_6
+      command = "#{@sudo} #{@puppetca} list --all"
+      logger.debug "Executing #{command}"
+      response = `#{command}`
+      unless $? == 0
+        logger.warn "Failed to run puppetca: #{response}"
+        raise "Execution of puppetca failed, check log files"
+      end
+
+      hash_name = "unknown"
+      hash = {}
+      valid = {}
+      revoked = {}
+      pending = {}
+
+      response.split("\n").each do |line|
+        if line.include? 'Signed Certificates:'
+          hash_name = 'valid'
+          next
+        elsif line.include? 'Requested Certificates:'
+          hash_name = 'pending'
+          next
+        elsif line.include? 'Revoked Certificates:'
+          hash_name = 'revoked'
+          next
+        end
+
+        (eval "#{hash_name}").merge! certificate_puppet_6(line, hash_name) rescue logger.warn("Failed to parse line: #{line}")
```

Comment:
Isn't there a way to avoid using `eval`? I don't see why you need to store `valid`, `revoked` and `pending` in separate hashes if they're going to be merged anyway. Otherwise I'd suggest something like:
```ruby
results = {
  'valid' => {},
  'revoked' => {},
  'pending' => {},
  'unknown' => {},
}
```